kotlinlib/src/mill/kotlinlib/js/KotlinJSModule.scala

```diff
@@ -379,16 +382,23 @@ trait KotlinJSModule extends KotlinModule { outer =>
 
   // these 2 exist to ignore values added to the display name in case of the cross-modules
   // we already have cross-modules in the paths, so we don't need them here
-  private def moduleName() = millModuleSegments.value
-    .filter(_.isInstanceOf[Segment.Label])
-    .map(_.asInstanceOf[Segment.Label])
-    .last
-    .value
+  private def moduleName() = {
+    val labelSegments = millModuleSegments.value
+      .filter(_.isInstanceOf[Segment.Label])
+      .map(_.asInstanceOf[Segment.Label])
```

Comment:
This can be simplified to millModuleSegments.value.collect{case label: Segment.Label => label}`. Same below